File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v1/service/UserService.java
package com.example.dtomappingbenchmark.v1.service;

import com.example.dtomappingbenchmark.v1.dto.UserDto;
import com.example.dtomappingbenchmark.v1.entity.User;
import com.example.dtomappingbenchmark.v1.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional
public class UserService {

    private final UserRepository userRepository;

    public UserDto create(UserDto dto){
        User entity = dto.toEntity();
        User saved = userRepository.save(entity);
        return saved.toDto();
    }

    public UserDto get(Long id) {
        return userRepository.findById(id).orElseThrow().toDto();
    }
}

File: object-mapping-benchmark-v4/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv4/user/service/UserService.kt
package yu.selab.cj.objectmappingbenchmarkv4.user.service


import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import yu.selab.cj.objectmappingbenchmarkv4.mapper.UserMapper
import yu.selab.cj.objectmappingbenchmarkv4.user.command.CreateUserDto
import yu.selab.cj.objectmappingbenchmarkv4.user.command.ReadUserDto
import yu.selab.cj.objectmappingbenchmarkv4.user.repository.UserRepository


@Service
class UserService(
    private val userRepository : UserRepository
) {

    @Transactional(readOnly = true)
    fun getUserList() = userRepository.findAll().map{
//        user -> ReadUserDto(null, null).fromEntity(user)
        user -> UserMapper.toDto(user)
    }

    @Transactional(readOnly = true)
    fun getUser(id : Long) : ReadUserDto {
        val user = userRepository.findById(id).orElseThrow {
            IllegalArgumentException("유저 없음")
        }

//        return ReadUserDto(null, null).fromEntity(user)
        return UserMapper.toDto(user)
    }

    @Transactional
    fun createUser(dto : CreateUserDto){
//        val user = dto.toEntity()
        val user = UserMapper.toEntity(dto)
        userRepository.save(user)
    }

    @Transactional
    fun deleteUser(id : Long)
        = userRepository.deleteById(id)

}